src/test/scala/chiselTests/reflect/DataMirrorSpec.scala

```diff
@@ -265,4 +265,37 @@ class DataMirrorSpec extends ChiselFlatSpec {
       foo.ports1 should be(Seq(foo.in, foo.out, foo.other))
     })
   }
+
+  "modulePorts and fullModulePorts" should "return an Instance of a module's IOs" in {
+    import chiselTests.experimental.hierarchy.Examples.AddOne
+    @instantiable
+    class Bar extends Module {
+      @public val io = IO(new Bundle {
+        val vec = Vec(2, Bool())
+        val x = UInt(4.W)
+      })
+      dontTouch(io)
+    }
+    class Foo extends Module {
+      val definition = Definition(new Bar)
+      val instA = Instance(definition)
+      val portsA = DataMirror.modulePorts(instA)
+
+      val instB = (Module(new Bar)).toInstance
+      val portsB = DataMirror.fullModulePorts(instB)
+    }
+
+    ChiselStage.emitCHIRRTL(new Module {
+      val foo = Module(new Foo)
+      foo.portsA.map(_._1) should contain("io")
+      foo.portsA.map(_._1) should not contain ("io_x")
+      (foo.portsB.map(_._1) should contain).allOf(
+        "io",
+        "io_x",
+        "io_vec",
+        "io_vec_0",
+        "io_vec_1"
+      )
```

Comment:
Can we provide a tighter check here?
```suggestion
      foo.portsA.map(_._1) should be(Seq("io"))
      (foo.portsB.map(_._1) should be(
        Seq(
          "io",
          "io_x",
          "io_vec",
          "io_vec_0",
          "io_vec_1"
        )
      )
```